packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { _formatString } from '@internal/acs-ui-common';
+import { RemoteParticipantState } from '@internal/calling-stateful-client';
+import { useEffect, useMemo, useState } from 'react';
+import { useLocale } from '../../localization';
+import { CallAdapterState } from '../adapter';
+import { useSelector } from '../hooks/useSelector';
+
+/**
+ * Custom selector for this hook to retrieve all the participants that are currently
+ * connected to the call.
+ */
+const getRemoteParticipantsConnectedSelector = (callState: CallAdapterState): RemoteParticipantState[] => {
+  return Object.values(callState.call?.remoteParticipants ?? {}).filter(
+    /**
+     * We need to make sure remote participants are in the connected state. If they are not
+     * they might not have their displayName set in the call state just yet.
+     */
+    (p) => p.state === 'Connected'
+  );
+};
+
+type ParticipantChangedAnnouncmentStrings = {
+  participantJoinedNoticeString: string;
+  twoParticipantJoinedNoticeString: string;
+  threeParticipantJoinedNoticeString: string;
+  participantLeftNoticeString: string;
+  twoParticipantLeftNoticeString: string;
+  threeParticipantLeftNoticeString: string;
+  unnamedParticipantChangedString: string;
+  manyParticipantsChanged: string;
+  manyUnnamedParticipantsChanged: string;
+};
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const strings = useMemo(() => {
```

Comment:
nitty nit: We usually just use the entire `locale` in the dependency array of hooks (below).
So you don't _really_ need this `useMemo`.

Either way is OK.